Meta/CMake/common_compile_options.cmake

```diff
@@ -5,17 +5,23 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_COLOR_DIAGNOSTICS ON)
 
-add_compile_options(-Wall)
-add_compile_options(-Wextra)
+if (MSVC)
+    add_compile_options(/W4)
+    # do not warn about unused function
+    add_compile_options(/wd4505)
+    # disable exceptions
+    add_compile_options(/EHsc)
+else()
+    add_compile_options(-Wall -Wextra)
+    add_compile_options(-fno-exceptions)
+    add_compile_options(-ffp-contract=off)
```

Comment:
`-ffp-contract=off` is also essential. We had some nasty bugs with floating point contraction enabled.

Reply:
Rebased and fixed in fe3c09eeca8d4050244f506a8eb99d4b9439f413